x/poolmanager/client/query_proto_wrap.go

```diff
@@ -324,30 +307,59 @@ func (q Querier) estimateTradeBasedOnPriceImpactInternal(
 		}
 	}
 
-	// First, try the full 'from coin' amount without a swap fee, if it errors it means there
-	// is very low liquidity for the trade we are attempting.
+	// Process the estimates according to the pool type.
+	switch poolI.GetType() {
+	case types.Balancer:
+		return q.estimateTradeBasedOnPriceImpactBalancerPool(
+			ctx, req, spotPrice, adjustedMaxPriceImpact, swapModule, poolI,
+		)
+	case types.Stableswap:
+		return q.estimateTradeBasedOnPriceImpactStableSwapPool(
+			ctx, req, spotPrice, adjustedMaxPriceImpact, swapModule, poolI,
+		)
+	case types.Concentrated:
+		return q.estimateTradeBasedOnPriceImpactConcentratedLiquidity(
+			ctx, req, spotPrice, adjustedMaxPriceImpact, swapModule, poolI,
+		)
+	default:
+		return nil, status.Error(codes.Internal, "pool not supported")
```

Comment:
It's the pool type that's not supported not the pool. And we should specify the pool type that's not supported.